spine-ts/spine-pixi/src/SpineTexture.ts

```diff
@@ -0,0 +1,109 @@
+import { BlendMode, Texture, TextureFilter, TextureWrap } from "@esotericsoftware/spine-core";
+import type { BaseTexture as PixiBaseTexture, BaseImageResource } from "@pixi/core";
+import { Texture as PixiTexture, SCALE_MODES, MIPMAP_MODES, WRAP_MODES, BLEND_MODES } from "@pixi/core";
+
+export class SpineTexture extends Texture {
+	private static textureMap: Map<PixiBaseTexture, SpineTexture> = new Map<PixiBaseTexture, SpineTexture>();
+
+	public static from(texture: PixiBaseTexture): SpineTexture {
+		if (SpineTexture.textureMap.has(texture)) {
+			return SpineTexture.textureMap.get(texture)!;
+		}
+		return new SpineTexture(texture);
+	}
+
+	public readonly texture: PixiTexture;
+
+	private constructor(image: PixiBaseTexture) {
+		// Todo: maybe add error handling if you feed a video texture to spine?
+		super((image.resource as BaseImageResource).source as any);
+		this.texture = PixiTexture.from(image);
+	}
+
+	public setFilters(minFilter: TextureFilter, _magFilter: TextureFilter): void {
+		this.texture.baseTexture.scaleMode = SpineTexture.toPixiTextureFilter(minFilter);
+		this.texture.baseTexture.mipmap = SpineTexture.toPixiMipMap(minFilter);
+
+		// pixi only has one filter for both min and mag, too bad
+	}
+
+	public setWraps(uWrap: TextureWrap, _vWrap: TextureWrap): void {
+		this.texture.baseTexture.wrapMode = SpineTexture.toPixiTextureWrap(uWrap);
+
+		// Pixi only has one setting
+	}
+
+	public dispose(): void {
+		// I am not entirely sure about this...
+		this.texture.destroy();
```

Comment:
there are 2 textures in pixi
`BaseTexture` holds the actual Image, HTMLCanvas, Compressed Texture or whatever is the actual data of your image.
`Texture` is a "view" on that BaseTexture. Mostly used to extract a rectangle or to set a pivot on a texture.

Destroying a Texture only affects that texture, but destroying a BaseTexture breaks all Textures that are a "view" of that.

Since Pixi doesn't support polygon packing and the spine runtime already handles the region extracting, I connected the Spine Texture directly to a Pixi Texture that its a view to the entire BaseTexture and let the runtime do the regioning during the Attachment processing.

Destroying `this.texture` is totally fine here. Destroying the underlaying BaseTexture (actually releasing the bytes that make up the image from memory) would mean that any other SpineTexture that depends on those same bytes will be broken too.

TLDR; 
Should "dispose" release completely the image data from memory? (if we ever need that image again we need to load the png again)